test/test_damper.cpp

```diff
@@ -0,0 +1,63 @@
+#include "teca_cf_reader.h"
+#include "teca_damper.h"
+#include "teca_l2_norm.h"
+#include "teca_connected_components.h"
+#include "teca_vtk_cartesian_mesh_writer.h"
+#include "teca_time_step_executive.h"
+#include "teca_system_interface.h"
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+int main(int argc, char **argv)
+{
+    teca_system_interface::set_stack_trace_on_error();
+
+    if (argc != 11)
+    {
+        cerr << "test_damper [dataset regex] "
+            "[x var] [y var] [z var] [t var] [u var] [v var] "
+            "[first step] [last step] [out file]"
+            << endl;
+        return -1;
+    }
+
+    string dataset_regex = argv[1];
+    string x_var = argv[2];
+    string y_var = argv[3];
+    string z_var = argv[4][0] == '.' ? "" : argv[4];
+    string t_var = argv[5][0] == '.' ? "" : argv[5];
+    string u_var = argv[6];
+    string v_var = argv[7];
+    int first_step = atoi(argv[8]);
+    int last_step = atoi(argv[9]);
+    string out_file = argv[10];
+
+    p_teca_cf_reader cfr = teca_cf_reader::New();
```

Comment:
intead of reading the data from a file, construct a mesh with a scalar field set to 1, and damp that. this will make it easier to validate the output.